waspc/src/Wasp/Generator/Valid/TsConfig.hs

```diff
@@ -1,112 +1,61 @@
-      validateRequiredFieldInCompilerOptions "isolatedModules" T.isolatedModules True,
-      validateRequiredFieldInCompilerOptions "jsx" T.jsx "preserve",
-      validateRequiredFieldInCompilerOptions "strict" T.strict True,
-      validateRequiredFieldInCompilerOptions "esModuleInterop" T.esModuleInterop True,
-      validateRequiredFieldInCompilerOptions "lib" T.lib ["dom", "dom.iterable", "esnext"],
-      validateRequiredFieldInCompilerOptions "allowJs" T.allowJs True,
-      -- Wasp internally uses TypeScript's project references to compile the
-      -- code. Referenced projects may not disable emit, so we must specify an
-      -- `outDir`.
-      validateRequiredFieldInCompilerOptions "outDir" T.outDir ".wasp/out/user",
-      -- The composite flag is required because Wasp uses project references
-      -- (i.e., web app and server reference user code as a subproject)
-      validateRequiredFieldInCompilerOptions "composite" T.composite True,
-      validateRequiredFieldInCompilerOptions "skipLibCheck" T.skipLibCheck True
-    ]
-  where
-    validateRequiredFieldInCompilerOptions relativeFieldName getFieldValue =
-      validateRequiredField (FieldName ["compilerOptions", relativeFieldName]) (getFieldValue compilerOptions)
-
-validateRequiredField :: (Eq a, JsonValue a) => FullyQualifiedFieldName -> Maybe a -> a -> [GeneratorError]
-validateRequiredField fullyQualifiedFieldName fieldValue expectedValue =
-  validateFieldValue fullyQualifiedFieldName (Just expectedValue) fieldValue
-
-validateFieldValue :: (Eq a, JsonValue a) => FullyQualifiedFieldName -> Maybe a -> Maybe a -> [GeneratorError]
-validateFieldValue fullyQualifiedFieldName expectedValue actualValue =
-  case (expectedValue, actualValue) of
-    (Nothing, Nothing) -> []
-    (Just expected, Just actual) -> [makeInvalidValueErrorMessage expected | actual /= expected]
-    (Just expected, Nothing) -> [makeMissingFieldErrorMessage expected]
-    (Nothing, Just _) -> [fieldMustBeUnsetErrorMessage]
+validateSrcTsConfig config =
+  GenericGeneratorError . show
+    <$> getValidationErrors validateFile config
   where
-    makeInvalidValueErrorMessage expected =
-      GenericGeneratorError $
-        unwords
-          [ "Invalid value for the",
-            "\"" ++ show fullyQualifiedFieldName ++ "\"",
-            "field in TS config, you must set it to:",
-            showAsJsValue expected ++ "."
+    -- References for understanding the required compiler options:
+    --   - The comments in templates/sdk/wasp/tsconfig.json
+    --   - https://www.typescriptlang.org/docs/handbook/modules/introduction.html
+    --   - https://www.totaltypescript.com/tsconfig-cheat-sheet
+    --   - https://www.typescriptlang.org/tsconfig/
+
+    validateFile =
+      withFileName "tsconfig.json" $
+        validateAll_
+          [ valueOfField ("include", T.include) $ eqJust ["src"],
+            valueOfField ("compilerOptions", T.compilerOptions) validateCompilerOptions
```

Comment:
Yeah, this line is a good example where `Satisfies` at the end of `valueOfField` helps with understanding.
Maybe `fieldValueSatisfies` might work hm.